apps/dashboard/src/main/java/com/akto/utils/GithubSync.java

```diff
@@ -0,0 +1,86 @@
+package com.akto.utils;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.akto.log.LoggerMaker;
+import com.akto.log.LoggerMaker.LogDb;
+
+public class GithubSync {
+    private static final Logger logger = LoggerFactory.getLogger(GithubSync.class);
+    private static final LoggerMaker loggerMaker = new LoggerMaker(GithubSync.class);
+    private static final OkHttpClient client = new OkHttpClient();
+
+    public String syncFile(String repo, String filePath) {
+        String fileContents = null; 
+
+        Request fileRequest = new Request.Builder()
+                            .url(String.format("https://raw.githubusercontent.com/%s/master/%s", repo, filePath))
+                            .build();
+                            
+        try {
+            Response fileResponse = client.newCall(fileRequest).execute();
+            ResponseBody fileResponseBody = fileResponse.body();
+            fileContents = fileResponseBody.string();
+        } catch (IOException ex) {
+            loggerMaker.errorAndAddToDb(String.format("Error while syncing file %s in Github repo %s %s ", repo, filePath, ex.toString()), LogDb.DASHBOARD);
+        }
+        
+        return fileContents;
+    }
+
+    public Map<String, String> syncDir(String repo, String dirPath) {
+        Map<String, String> dirContents = new HashMap<>();
+
+        // Get Github repo tree - Eg: https://api.github.com/repos/akto-api-security/akto/git/trees/master?recursive=1
+        Request treeRequest = new Request.Builder()
+                .url(String.format("https://api.github.com/repos/%s/git/trees/master?recursive=1", repo))
+                .build();
```

Comment:
use sync file func inside sync dir, so that we have file fetching at only 1 place